Data/ByteString/Lazy.hs

```diff
@@ -696,6 +725,28 @@ drop i cs0 = drop' i cs0
             then Chunk (S.drop (fromIntegral n) c) cs
             else drop' (n - fromIntegral (S.length c)) cs
 
+-- | /O(c)/ @'dropEnd' n xs@ is equivalent to @'take' ('length' xs - n) xs@.
+-- Drops @n@ elements from end of bytestring.
+--
+-- >>> dropEnd 3 "abcdefg"
+-- "abcd"
+-- >>> dropEnd 0 "abcdefg"
+-- "abcdefg"
+-- >>> dropEnd 4 "abc"
+-- ""
+--
+-- @since 0.11.2.0
+dropEnd :: Int64 -> ByteString -> ByteString
+dropEnd i p | i <= 0 = p
+dropEnd i cs0 = dropEnd' i cs0
```

Comment:
I suggest we make `dropEnd` lazier in a fashion vaguely similar to `dropWhileEnd`. Imagine `dropEnd 10` and chunks of size 1, 2, 3, 6... As soon as you encountered the fourth chunk, you know that you can output the first one no matter what. Because 1 <= 2 + 3 + 6 - 10.

Reply:
Hi @Bodigrim , `dropEnd` updated. I'm slowly  getting there 😄